modules/brushingandlinking/src/brushingandlinkingmanager.cpp

```diff
@@ -31,78 +31,422 @@
+    throw Exception("clearing filtered indices is no longer supported", IVW_CONTEXT);
+}
+
+void BrushingAndLinkingManager::clearSelected(BrushingTarget target) {
+    clearIndices(BrushingAction::Select, target);
+}
+
+void BrushingAndLinkingManager::clearHighlighted(BrushingTarget target) {
+    clearIndices(BrushingAction::Highlight, target);
+}
+
+size_t BrushingAndLinkingManager::getNumberOfFiltered(BrushingTarget target) const {
+    return getIndices(BrushingAction::Filter, target).size();
+}
+
+size_t BrushingAndLinkingManager::getNumberOfSelected(BrushingTarget target) const {
+    return getIndices(BrushingAction::Select, target).size();
+}
+
+size_t BrushingAndLinkingManager::getNumberOfHighlighted(BrushingTarget target) const {
+    return getIndices(BrushingAction::Highlight, target).size();
+}
+
+const BitSet& BrushingAndLinkingManager::getFilteredIndices(BrushingTarget target) const {
+    return getIndices(BrushingAction::Filter, target);
+}
 
-size_t BrushingAndLinkingManager::getNumberOfSelected() const { return selected_.size(); }
+const BitSet& BrushingAndLinkingManager::getSelectedIndices(BrushingTarget target) const {
+    return getIndices(BrushingAction::Select, target);
+}
 
-size_t BrushingAndLinkingManager::getNumberOfFiltered() const { return filtered_.getSize(); }
+const BitSet& BrushingAndLinkingManager::getHighlightedIndices(BrushingTarget target) const {
+    return getIndices(BrushingAction::Highlight, target);
+}
 
-size_t BrushingAndLinkingManager::getNumberOfSelectedColumns() const {
-    return selectedColumns_.size();
+bool BrushingAndLinkingManager::isFiltered(uint32_t idx, BrushingTarget target) const {
+    return contains(idx, BrushingAction::Filter, target);
 }
 
-void BrushingAndLinkingManager::remove(const BrushingAndLinkingInport* src) {
-    filtered_.remove(src);
+bool BrushingAndLinkingManager::isSelected(uint32_t idx, BrushingTarget target) const {
+    return contains(idx, BrushingAction::Select, target);
 }
 
-bool BrushingAndLinkingManager::isColumnSelected(size_t idx) const {
-    return selectedColumns_.find(idx) != selectedColumns_.end();
+bool BrushingAndLinkingManager::isHighlighted(uint32_t idx, BrushingTarget target) const {
+    return contains(idx, BrushingAction::Highlight, target);
 }
 
-void BrushingAndLinkingManager::setSelected(const BrushingAndLinkingInport*,
-                                            const std::unordered_set<size_t>& indices) {
-    selected_ = indices;
-    owner_->invalidate(invalidationLevel_);
+void BrushingAndLinkingManager::setParent(BrushingAndLinkingManager* parent) {
+    if (parent_) {
+        parent_->removeChild(this);
+
+        // set manager as modified since filter actions might have changed when removing the parent
+        modifiedActions_ =
+            BrushingModifications(BrushingModification::Filtered, BrushingModification::Selected,
+                                  BrushingModification::Highlighted);
```

Comment:
```suggestion
        modifiedActions_ = BrushingModifications(flags::any);
```
If my memory serves me right